src/Features/Lsif/Generator/Generator.cs

```diff
@@ -400,6 +403,29 @@ void MarkImplementationOfSymbol(ISymbol baseMember)
             }
         }
 
+        private static DefinitionRangeTag? CreateRangeTagForDeclaredSymbol(ISymbol declaredSymbol, SyntaxTree syntaxTree, ISyntaxFacts syntaxFacts, CancellationToken cancellationToken)
+        {
+            // Find the syntax node that declared the symbol in the tree we're processing
+            var syntaxReference = declaredSymbol.DeclaringSyntaxReferences.FirstOrDefault(static (r, syntaxTree) => r.SyntaxTree == syntaxTree, arg: syntaxTree);
+            var syntaxNode = syntaxReference?.GetSyntax(cancellationToken);
+
+            if (syntaxNode is null)
+                return null;
+
+            // The containing range is supposed to be "the full range of the declaration not including leading/trailing whitespace, but everything else", so produce our start/end points
+            // looking through trivia
+            var firstNonWhitespaceTrivia = syntaxNode.GetLeadingTrivia().FirstOrNull(static (t, syntaxFacts) => !syntaxFacts.IsWhitespaceOrEndOfLineTrivia(t), syntaxFacts);
+            var fullRangeStart = firstNonWhitespaceTrivia?.SpanStart ?? syntaxNode.SpanStart;
+
+            var lastNonWhitespaceTrivia = syntaxNode.GetTrailingTrivia().LastOrNull(static (t, syntaxFacts) => !syntaxFacts.IsWhitespaceOrEndOfLineTrivia(t), syntaxFacts);
+            var fullRangeEnd = lastNonWhitespaceTrivia?.Span.End ?? syntaxNode.Span.End;
+
+            var fullRange = ProtocolConversions.TextSpanToRange(TextSpan.FromBounds(fullRangeStart, fullRangeEnd), syntaxTree.GetText(cancellationToken));
+            var symbolKind = ProtocolConversions.GlyphToSymbolKind(declaredSymbol.GetGlyph());
+
+            return new DefinitionRangeTag(declaredSymbol.Name, symbolKind, fullRange);
```

Comment:
I'm using declaredSymbol.Name here rather than the true "token text" which would be escaped; I don't imagine that's useful but @gundermanc or @noellelc what do you actually need here?

Reply:
Can you give me an example of what `declaredSymbol.Name` is, versus the "token text" you mention? Just want to make sure I'm understanding the question before answering.

Reply:
C# allows for escaping of identifiers. For example if you really want your class to be called "class", you can escape with an @:

```
class @class
{
}
```

Do you want in this case the name to be "class" or "@class"? The spec implies I should perhaps be returning the latter (since that's the text of the span) but that may not be useful depending on your needs.

Reply:
I think we would want to include the @ sign as most uses of documentSymbol, like symbol navigation, would presumably need that.

Reply:
Done. and I think that more correctly matches the spec either way.